### PR TITLE
Add support for Vim-Signify

### PR DESCRIPTION
**Pull request by hkupty**
As well as gitgutter, add support for vim-signify.

### CODE REVIEW

**Comment by jreybert** [comment]
Thanks for the PR!

But jenkins is positive:

> Error detected while processing /home/travis/build/jreybert/vimagit/plugin/magit.vim:
> line   74:
> E580: :endif without :if: endif

Please fix your commit, preferably with a `commit --amend` (if error is in last commit) or a `rebase --autosquash`

**Comment by hkupty** [comment]
Sorry about that.. It was a bad leftover from previous commit.. Now it should work.

File: plugin/magit.vim [hunk 1]
```diff
@@ -50,26 +50,25 @@ let g:magit_default_fold_level     = get(g:, 'magit_default_fold_level',
...
 
-if ( g:magit_refresh_gitgutter == 1 )
-	autocmd User VimagitUpdateFile
-			\ if ( exists("*gitgutter#process_buffer") ) |
-			\ 	call gitgutter#process_buffer(bufnr(g:magit_last_updated_buffer), 0) |
-			\ endif
-endif
+if ( g:magit_refresh_gutter == 1 || g:magit_refresh_gitgutter == 1)
+
+  if ( exists("*gitgutter#process_buffer") )
```
Comment (jreybert) [inline plugin/magit.vim]: You save a if statement at runtime, but the evaluation is done at plugin initialization. It means that if vimagit is loaded before gitgutter and/or vim-signify, `exists("*gitgutter#process_buffer")` will be evaluated to false.

I do not like either the `if` in the autocmd, but this is the only way to not be dependent on plugin loading order.

Please, only add vim signify support for this PR.

If you find a better way to check gitgutter/vim-signify presence, I'll be happy to merge :)

**Review by jreybert — CHANGES_REQUESTED** [review]

File: plugin/magit.vim (hunk 1 above)
Comment (hkupty) [reply plugin/magit.vim]: Sure.. That makes sense.

I'll cut this down to solving signify for now.

Aside from the if controversy, are you ok with changing the variable name?

**Comment by jreybert** [comment]
Excellent!

Could you please update the doc/vimagit.txt and README.md with the new `g:magit_refresh_gutter` and add a note for the deprecated one.

Thanks again for the PR!

**Comment by jreybert** [comment]
Argh, I missed the jenkins complaint!

> Vim(if):E15: Invalid expression: (exists("_gitgutter#process_buffer"))   call gitgutter#process_buffer(bufnr(g:magit_last_updated_buffer), 0) elseif (exists("_sy#util#refresh_windows"))   call sy#util#refresh_windows() endif

I'll do the doc update, just fix this for the merge.

**Comment by hkupty** [comment]
Sorry for taking long. This should be fixed now.

**Comment by hkupty** [comment]
Can we merge this?

**Review by jreybert — APPROVED** [review]

**Comment by jreybert** [comment]
Hi Henry,

I'm very sorry for the delay. I really don't have much time at the moment. It's a pity because you did two nice PRs. You're also the first one to propose PRs touching in the heart of vimagit. That's why I am so careful.

The content of this PR is OK to me. I don't want to discourage you (I really should write some PR guideline). Could you:
- make the pull request on branch `next` (mandatory)
- reduce the number of commits (nice to have)

For the first point, I want to test your feature one week or two before merging it to master. It is mandatory.

For the second point, it is a nice to have for this commit. It will be mandatory for your next PRs. In a PR, I would prefer to not have:
- add feature foo
- fix feature foo
  Please use `git commit --fixup` and `git rebase --autosquash`.

Thanks for your effort!

**Comment by hkupty** [comment]
No problem!

I'll address those points later this week, as I'm out on holidays.

**Comment by hkupty** [comment]
@jreybert Done

**Comment by jreybert** [comment]
Thanks!

**Pull request merged**